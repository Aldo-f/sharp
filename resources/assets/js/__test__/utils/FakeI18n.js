export default {
    install() {
        window.i18n = {};
    }
}