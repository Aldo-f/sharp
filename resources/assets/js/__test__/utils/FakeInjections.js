import Vue from 'vue';

const injectedComponents = ['$field', '$form'];

export default {
    provide() {
        return injectedComponents.reduce((injections, prop) => {
            injections[prop] = new Vue();
            return injections;
        }, {});
    }
}