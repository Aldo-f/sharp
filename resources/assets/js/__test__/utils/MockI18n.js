import Localization ,* as localizationFn from '../../mixins/Localization';

const mockLangImplementation = localeKey => `{{ ${localeKey} }}`;

export default {
    install() {
        window.i18n = {};
    },

    mockLangFunction() {
        localizationFn.lang = jest.fn(mockLangImplementation);
        Localization.methods.l = jest.fn(mockLangImplementation);
    }
}