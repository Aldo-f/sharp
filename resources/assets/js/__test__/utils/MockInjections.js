import Vue from 'vue';

const injectedComponents = {
    $field:{},
    $form:{
        data: () => ({
            errors: {}
        })
    },
    $tab:{}
};

export default {
    provide() {
        return Object.keys(injectedComponents).reduce((injections, compName) => {
            let Comp = Vue.extend(injectedComponents[compName]);
            injections[compName] = new Comp();
            return injections;
        }, {});
    }
}