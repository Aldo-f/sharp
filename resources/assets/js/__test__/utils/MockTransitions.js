export default {
    install() {
        window.getComputedStyle = jest.fn(() => ({
            animationDelay:'',
            animationDuration:'',
            transitionDelay:'',
            transitionDuration:''
        }));
    }
}