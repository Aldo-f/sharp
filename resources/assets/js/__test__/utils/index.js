export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC } from './mockSFC';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}