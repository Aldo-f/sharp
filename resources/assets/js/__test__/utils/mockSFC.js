export function mockSFC (sfcModule, { template, render, ...extend }={}, mockOptions={}) {

    let { isJsFile } = mockOptions;

    if(isJsFile) {
        sfcModule.__oldDefault__ = { ...sfcModule.default }
    }

    Object.assign(sfcModule.default, {
        mixins: [],
        'extends': extend,
        template: template,
        render: !template ? render || function() { return this._v(`__MOCKED_SFC_${sfcModule.name}__\n`) } : null,
        watch: {},
        beforeCreate: ()=>{},
        created: ()=>{},
        beforeMount: ()=>{},
        mounted: ()=>{},
        beforeUpdate: ()=>{},
        updated: ()=>{},
        beforeDestroy: ()=>{},
        destroyed: ()=>{},
        data: ()=>({}),
        provide: {},
        inject: [],
    });

    //console.log(sfcModule);
    return sfcModule.default;
}

export function unmockSFC(sfcModule) {
    if(sfcModule.__oldDefault__) {
        sfcModule.default = sfcModule.__oldDefault__;
        return;
    }

    let { 'default':defaultObj, ...options } = sfcModule;
    sfcModule.default = options;
}