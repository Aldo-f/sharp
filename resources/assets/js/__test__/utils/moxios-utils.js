import moxios from 'moxios';

export function nextRequestFulfilled(config) {
    return new Promise(resolve => {
        moxios.wait(async () => {
            let request = moxios.requests.mostRecent();
            let response = await request.respondWith(config);
            resolve(response); // moxios Response object
        })
    });
}
