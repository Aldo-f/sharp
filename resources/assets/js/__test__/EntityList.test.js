import Vue from 'vue';
import EntityList from '../components/list/EntityList.vue';
import Dropdown from '../components/dropdown/Dropdown.vue';
import Modal from '../components/Modal.vue';

import * as consts from '../consts';

import { mockChildrenComponents } from "./utils/mockSFC";

import { mockProperty, unmockProperty, setter } from "./utils/mock-utils";

import moxios from 'moxios';
import {MockInjections, MockI18n} from "./utils";
import { nextRequestFulfilled } from './utils/moxios-utils';


import mocks from './__mocks__/entityList';


describe('entity-list', ()=>{
    Vue.use(MockI18n);
    MockI18n.mockLangFunction();

    Vue.component('sharp-entity-list', mockChildrenComponents(EntityList, {
        customs: {
            [Dropdown.name]: {
                template: '<div class="MOCKED_SHARP_DROPDOWN"><slot></slot></div>'
            },
            'Draggable': {
                template: '<div id="MOCKED_VUE_DRAGGALBE"><slot></slot></div>'
            },
            [Modal.name]: {
                template: '<div class="MOCKED_SHARP_MODAL"><slot></slot></div>'
            }
        }
    }));

    moxios.delay = 10;

    beforeEach(()=>{
        document.body.innerHTML = `
            <div id="app">
                <sharp-entity-list entity-key="spaceship"></sharp-entity-list>
            </div>
        `;
    });
    afterEach(()=>{
        moxios.uninstall();
    });

    it('can mount entity list', async ()=>{
        await createVm();

        await nextRequestFulfilled({
            status: 200,
            response: mocks.basicSingleContainerFullAuthorizations
        });

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm(customOptions={}) {

    const vm = new Vue({
        el: '#app',
        mixins: [MockInjections, customOptions],

        props: ['independant', 'props'],

        'extends': {
            data:()=>({
                instanceId: null,
                ignoreAuthorizations: null
            })
        },

        created() {
            let { axiosInstance } = this._provided;
            moxios.install(axiosInstance);
            moxios.uninstall = moxios.uninstall.bind(moxios, axiosInstance);
        }
    });

    await Vue.nextTick();

    return vm.$children[0];
}