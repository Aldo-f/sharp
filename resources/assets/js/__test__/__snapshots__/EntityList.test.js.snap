// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entity-list can mount entity list 1`] = `
<div id="app">
    <div class="SharpEntityList">
        <div class="SharpEntityList__table SharpEntityList__table--border">
            <div class="SharpEntityList__thead">
                <div class="SharpEntityList__row SharpEntityList__row--header container">
                    <div class="SharpEntityList__cols">
                        <div class="row">
                            <div class="SharpEntityList__th col-12 col-md-6"><span>Titre</span>
                                <!---->
                                <!---->
                            </div>
                        </div>
                    </div>
                    <div class="d-none d-md-block" style="width: 0px;">&nbsp;</div>
                </div>
            </div>
            <div class="SharpEntityList__tbody">
                <div id="MOCKED_VUE_DRAGGALBE">
                    <div class="SharpEntityList__row container">
                        <div class="SharpEntityList__cols">
                            <div class="row">
                                <div class="SharpEntityList__td col-12 col-md-6">
                                    Super title
                                </div>
                            </div> <a href="/sharp/form/spaceship/1" class="SharpEntityList__row-link"></a></div>
                        <div class="SharpEntityList__row-actions">
                            <!---->
                            <!---->
                        </div>
                        <div class="SharpEntityList__row-actions" style="display: none;"><i class="fa fa-ellipsis-v SharpEntityList__reorder-icon"></i></div>
                    </div>
                </div>
            </div>
        </div>
        <div class="SharpEntityList__pagination-container">
            <!---->
        </div> __MOCKED_SFC_SharpViewPanel__
    </div>
</div>
`;
